### ISSUE TITLE
Windows Computer Object - Investigate Invalid Content Type

### ISSUE THREAD

**Issue by rkemery-hashicorp** [hashicorp/terraform-provider-ad#41]
### Terraform Version and Provider Version
Terraform version: 0.13
Provider version: terraform-provider-ad_v0.1.0_x5

### Windows Version
Server 2019 Standard

### Affected Resource(s)
```hcl
resource "ad_computer" "c" {
name = "test"
pre2kname = "TEST"
}  
```

### Terraform Configuration Files
```hcl
provider "ad" {
winrm_hostname = ""
winrm_password = ""
winrm_username = ""
}

resource "ad_computer" "c" {
name = "test"
pre2kname = "TEST"
# container = ""
} 
```

### Debug Output
https://gist.github.com/rkemery-hashicorp/00a5c0b74ee706791e4514d7f47c6b08

### Expected Behavior
Computer object is created.

### Actual Behavior
```
Error: error while creating new computer object: winrm execution failure while creating computer object: powershell command failed with exit code 1
stdout:
stderr:
```

### Steps to Reproduce
Create terraform config with sameple config to add computer object.

### Important Factoids
None

### References
None

### Community Note
* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by elongstreet88** [comment]
Also receiving this with Terraform v0.12.29

**Comment by rkemery-hashicorp** [comment]
Appears to work if you enable basic auth in winrm:

`winrm set winrm/config/service/auth @{Basic="true"}`

**Comment by elongstreet88** [comment]
> Appears to work if you enable basic auth in winrm:
> 
> `winrm set winrm/config/service/auth @{Basic="true"}`

Did you have to set AllowUnencrypted set as well?

If this provider doesn't support Kerberos, Negotiate, or one of the other more secure mechanisms, it may be tough to adopt as basic + unencrypt == password on the wire.

**Comment by rkemery-hashicorp** [comment]
I set AllowUnencrypted = true but didn't test extensively with that alone.

**Comment by elongstreet88** [comment]
Based on winrm requirements for other Go/terraform providers/provisioners I've passed by, i wouldn't be surprised if this is the case.

I would think at a minimum, these needs to be called out in the provider documentation. Or if i'm off base, provide info about what IS supported if its more than basic+allowunencrypted.

For folks like me, this is a non-starter for using this fantastic provider concept.

**Comment by elongstreet88** [comment]
Looking at the downstream winrm provider [https://github.com/masterzen/winrm], it looks like the following restrictions are required:
```
Preparing the remote Windows machine for Basic authentication
This project supports only basic authentication for local accounts (domain users are not supported). The remote windows system must be prepared for winrm:

For a PowerShell script to do what is described below in one go, check Richard Downer's blog

On the remote host, a PowerShell prompt, using the Run as Administrator option and paste in the following lines:

	winrm quickconfig
	y
	winrm set winrm/config/service/Auth '@{Basic="true"}'
	winrm set winrm/config/service '@{AllowUnencrypted="true"}'
	winrm set winrm/config/winrs '@{MaxMemoryPerShellMB="1024"}'
```

If thats the case, no kerberos or other support for the time being and plain text passwords on the wire :\.

**Comment by aareet** [comment]
For the technical preview, we wanted to start with basic auth. We'll investigate whether it's possible to support Kerberos or other more secure forms of authentication. We currently have https://github.com/hashicorp/terraform-provider-ad/issues/35 filed as well.

**Comment by elongstreet88** [comment]
> For the technical preview, we wanted to start with basic auth. We'll investigate whether it's possible to support Kerberos or other more secure forms of authentication. We currently have #35 filed as well.

Thanks for the update. I'm sure this one will be a pain to implement, but like I said, its really a non-starter (or probably SHOULD be) for a lot of people due to the plain text password on the wire.

I'm sure rebuilding a winrm provider will be a blast.
/s

Looking forward to seeing this grow!

**Comment by petercharleston** [comment]
Any update on this provider working with HTTPS?

**Comment by ibauersachs** [comment]
HTTPS is working fine if you enable it on the DC (e.g. with `winrm quickconfig -transport:https`, don't forget to open port 5986 on the firewall) and configure the provider:
```
provider "ad" {
  winrm_hostname = "..."
  winrm_insecure = false
  winrm_port = 5986
  winrm_proto = "https"
  winrm_username = "..."
  winrm_password = "..."
}
```

**Comment by faisaleem-na** [comment]
any update on this issue?
using HTTPS but basic auth and allowunencrypted is disabled via GPO, so get this same error

**Comment by koikonom** [comment]
@faisaleem-na have you tried authenticating via Kerberos?  https://registry.terraform.io/providers/hashicorp/ad/latest/docs#note-about-kerberos-authentication

**Comment by jok0** [comment]
> @faisaleem-na have you tried authenticating via Kerberos? https://registry.terraform.io/providers/hashicorp/ad/latest/docs#note-about-kerberos-authentication

For anyone else struggling to connect these up.  This worked for me.

**Comment by faisaleem-na** [comment]
> > @faisaleem-na have you tried authenticating via Kerberos? https://registry.terraform.io/providers/hashicorp/ad/latest/docs#note-about-kerberos-authentication
> 
> For anyone else struggling to connect these up. This worked for me.
> 
> Thanks @koikonom

Yep works for me too after using the kerberos realm env variable
